test/shadertools/lib/shader-cache.spec.js

```diff
@@ -138,3 +139,33 @@ test('Experimental#ShaderCache - construct Program from cached shaders', t => {
 
   t.end();
 });
+
+test('Experimental#ShaderCache - check recycling with cache', t => {
```

Comment:
This looks good. Some ideas:
* Add a test showing that a second call generates a new `Program`?
* Possibly add a test showing that a program with parameters we can't handle (varying etc) doesn't get cached.